spec/PasswordPolicy.spec.js

```diff
@@ -757,4 +759,233 @@ describe("Password Policy: ", () => {
+    var user = new Parse.User();
+    var emailAdapter = {
+      sendVerificationEmail: () => Promise.resolve(),
+      sendPasswordResetEmail: options => {
+        requestp.get({
+          uri: options.link,
+          followRedirect: false,
+          simple: false,
+          resolveWithFullResponse: true
+        }).then(response => {
+          expect(response.statusCode).toEqual(302);
+          var re = /http:\/\/localhost:8378\/1\/apps\/choose_password\?token=([a-zA-Z0-9]+)\&id=test\&username=user1/;
+          var match = response.body.match(re);
+          if (!match) {
+            fail("should have a token");
+            done();
+            return;
+          }
+          var token = match[1];
+
+          requestp.post({
+            uri: "http://localhost:8378/1/apps/test/request_password_reset",
+            body: `new_password=uuser11&token=${token}&username=user1`,
+            headers: {
+              'Content-Type': 'application/x-www-form-urlencoded'
+            },
+            followRedirect: false,
+            simple: false,
+            resolveWithFullResponse: true
+          }).then(response => {
+            expect(response.statusCode).toEqual(302);
+            expect(response.body).toEqual('Found. Redirecting to http://localhost:8378/1/apps/password_reset_success.html?username=user1');
+
+            Parse.User.logIn("user1", "uuser11").then(function (user) {
+              done();
+            }).catch(err => {
+              jfail(err);
+              fail("should login with new password");
+              done();
+            });
+
+          }).catch(error => {
+            jfail(error);
+            fail("Failed to POST request password reset");
+          });
+        }).catch(error => {
+          jfail(error);
+          fail("Failed to get the reset link");
+        });
+      },
+      sendMail: () => {
+      }
+    };
+    reconfigureServer({
+      appName: 'passwordPolicy',
+      emailAdapter: emailAdapter,
+      passwordPolicy: {
+        daysBeforeExpiry: 0.000005787 // 0.5 sec
+      },
+      publicServerURL: "http://localhost:8378/1"
+    }).then(() => {
+      user.setUsername("user1");
+      user.setPassword("user1");
+      user.set('email', 'user1@parse.com');
+      user.signUp().then(() => {
+        // wait for a bit more than the validity duration set
+        setTimeout(() => {
+          Parse.User.logIn("user1", "user1").then(() => {
+            fail("logIn should have failed");
+            done();
+          }).catch((error) => {
+            expect(error.code).toEqual(Parse.Error.OBJECT_NOT_FOUND);
+            expect(error.message).toEqual('Your password has expired. Please reset your password.');
+            Parse.User.requestPasswordReset('user1@parse.com', {
+              error: (err) => {
+                jfail(err);
+                fail("Reset password request should not fail");
+                done();
```

Comment:
@bhaskaryasa 
I don't see you checking for the `reset password timestamp`
Can you pls check the DB to make sure the timestamp for the password has been updated?